src/libraries/System.IO.FileSystem/tests/FileStream/FileStreamConformanceTests.cs

```diff
@@ -274,4 +278,145 @@ protected override async Task<StreamPair> CreateConnectedStreamsAsync()
     }
+
+    [PlatformSpecific(TestPlatforms.Windows)] // DOS device paths (\\.\ and \\?\) are a Windows concept
+    public class UnseekableDeviceFileStreamConnectedConformanceTests : ConnectedStreamConformanceTests
+    {
+        protected override async Task<StreamPair> CreateConnectedStreamsAsync()
+        {
+            string pipeName = FileSystemTest.GetNamedPipeServerStreamName();
+            string pipePath = Path.GetFullPath($@"\\.\pipe\{pipeName}");
+
+            var server = new NamedPipeServerStream(pipeName, PipeDirection.In);
+            var clienStream = new FileStream(File.OpenHandle(pipePath, FileMode.Open, FileAccess.Write, FileShare.None), FileAccess.Write);
+
+            await server.WaitForConnectionAsync();
+
+            var serverStrean = new FileStream(new SafeFileHandle(server.SafePipeHandle.DangerousGetHandle(), true), FileAccess.Read);
+
+            server.SafePipeHandle.SetHandleAsInvalid();
+
+            return (serverStrean, clienStream);
+        }
+
+        protected override Type UnsupportedConcurrentExceptionType => null;
+        protected override bool UsableAfterCanceledReads => false;
+        protected override bool FullyCancelableOperations => false;
+        protected override bool BlocksOnZeroByteReads => OperatingSystem.IsWindows();
+        protected override bool SupportsConcurrentBidirectionalUse => false;
+    }
+
+    [PlatformSpecific(TestPlatforms.Windows)] // DOS device paths (\\.\ and \\?\) are a Windows concept
+    public class SeekableDeviceFileStreamStandaloneConformanceTests : UnbufferedAsyncFileStreamStandaloneConformanceTests
+    {
+        protected override string GetTestFilePath(int? index = null, [CallerMemberName] string memberName = null, [CallerLineNumber] int lineNumber = 0)
+        {
+            string filePath = Path.GetFullPath(base.GetTestFilePath(index, memberName, lineNumber));
+            string drive = Path.GetPathRoot(filePath);
+            StringBuilder volumeNameBuffer = new StringBuilder(filePath.Length + 1024);
+
+            // the following method maps drive letter like "C:\" to a DeviceID (a DOS device path)
+            // example: "\\?\Volume{724edb31-eaa5-4728-a4e3-f2474fd34ae2}\"
+            if (!GetVolumeNameForVolumeMountPoint(drive, volumeNameBuffer, volumeNameBuffer.Capacity))
+            {
+                throw new Win32Exception(Marshal.GetLastPInvokeError(), "GetVolumeNameForVolumeMountPoint failed");
+            }
+
+            // instead of:
+            // 'C:\Users\x\AppData\Local\Temp\y\z
+            // we want sth like:
+            // '\\.\Volume{724edb31-eaa5-4728-a4e3-f2474fd34ae2}\Users\x\AppData\Local\Temp\y\z
+            string devicePath = filePath.Replace(drive, volumeNameBuffer.ToString());
+            Assert.StartsWith(@"\\?\", devicePath);
+#if DEBUG
+            devicePath = devicePath.Replace(@"\\?\", @"\\.\"); // we want to test both \\.\ and \\?\
```

Comment:
I assume the comment is referring to testing one in debug and the other in release?